packages/adaptive_scaffold/lib/src/adaptive_layout.dart

```diff
@@ -0,0 +1,517 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui';
+import 'package:flutter/widgets.dart';
+import 'breakpoints.dart';
+import 'slot_layout.dart';
+
+enum _SlotIds {
+  primaryNavigation,
+  secondaryNavigation,
+  topNavigation,
+  bottomNavigation,
+  body,
+  secondaryBody,
+}
+
+/// [AdaptiveLayout] is dedicated to simplifying the process of developing
+/// layouts for many different screen sizes. It separates the screen into
+/// several sections, also referred to as "slots". The Widget defines slots for
+/// the different placements of navigational elements and two slots for a body
+/// and secondaryBody. All of these slots take a [SlotLayout] Widget. This class
+/// works in junction with [SlotLayout] and [SlotLayoutConfig]. This class
+/// handles the placement of the slots on the screen and animations regarding
+/// their macromovements.
+///
+/// See also:
+///
+///  * [SlotLayout], which handles the actual switching and animations between
+/// elements based on [Breakpoint]s.
+///  * [SlotLayoutConfig], which holds information regarding the actual Widgets
+/// and the desired way to animate between switches. Often used within
+/// [SlotLayout].
+///  * [Design Doc](https://flutter.dev/go/adaptive-layout-foldables).
+///  * [Material Design 3 Specifications] (https://m3.material.io/foundations/adaptive-design/overview).
+class AdaptiveLayout extends StatefulWidget {
+  /// Creates an [AdaptiveLayout] widget.
+  const AdaptiveLayout({
+    this.primaryNavigation,
+    this.secondaryNavigation,
+    this.topNavigation,
+    this.bottomNavigation,
+    this.body,
+    this.secondaryBody,
+    this.bodyRatio,
+    this.internalAnimations = true,
+    this.bodyOrientation = Axis.horizontal,
+    super.key,
+  });
+
+  /// The slot placed on the beginning side of the screen; meaning on the left
```

Comment:
No need to reiterate "beginning" (since it's in the name). Might as well just say: the slot that appears on the left when [Directionality.textDirection] is LTR, and on the right for RTL.

This slot's name (and many others) have a navigation suffix. Do we need to explain that the slot typically holds a navigational element, like a AppBar?

Reply:
I think it helps to have beginning and end since primary and secondary don't necessarily imply their placement. 
The navigation suffix is to nudge the developer to place some sort of navigational elements in that area. But really anything can be placed in these slots.

Reply:
Sorry, I think I mixed this up with something else.